### PR TITLE
All changes we discussed

### PR DESCRIPTION
**Pull request by Igor-Arena**
1. Changed files structure in the git:
- deleted duplicated files
- moved Api tests into test >  Api_tests folder
- created "Global_settings" folder in the "tests" folder for all global files
- moved my own files for studying into "automation_cource" folder
2. Changed Var to the Let for all tests
3. Names of variables are written from the small first letter according to the convention
4. Used global variables for api url 
5. Used "import..." in all files for import
6. Changed hardcoded data to the dynamic using "faker"
7. Removed "console.log" where its not needed

### CODE REVIEW

File: tests/Api_tests/post.spec.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  ],
+  "tags": [
+    {
+      "id": 0,
+      "name": "string"
+    }
+  ],
+  "status": "available"
+ };
+
```
Comment (siarheima-papaya) [inline tests/Api_tests/post.spec.ts:29]: add test.afterAll() and remove created pet. It necessary to remove test data after test launch. We always should clean up test data after tests

File: tests/Api_tests/post.spec.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import {test, expect} from '@playwright/test'
```
Comment (siarheima-papaya) [inline tests/Api_tests/post.spec.ts:1]: All comments above valid for other tests. Please, fix them

File: tests/Api_tests/delete.spec.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
+import {test, expect} from '@playwright/test'
+import axios from 'axios';
+import {faker, simpleFaker} from '@faker-js/faker';
+import { postAndGetPet } from '../Global_settings/fake_global_values';
+
+
+let petId: number;
+let randomFirstName = faker.person.firstName(undefined);
```
Comment (siarheima-papaya) [inline tests/Api_tests/delete.spec.ts:8]: remove 'undefined'. You can use faker.person.firstName()

File: package.json [hunk 4]
```diff
@@ -8,6 +8,7 @@
   "author": "",
```
Comment (siarheima-papaya) [inline package.json:8]: Please, add script to run your tests

File: tests/Api_tests/put.spec.ts [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+        "tags": [
+          {
+            "id": 0,
+            "name": "string"
+          }
+        ],
+        "status": "available"}
+    });
+    console.log(response);
+    expect(response.status).toBe(200);
```
Comment (siarheima-papaya) [inline tests/Api_tests/put.spec.ts:63]: Add assertion to validate that data was changed

**Comment by Igor-Arena** [comment]
1. Changed name of folders according to the convention
2. Added script
3. Added "Delete" api for tests
4. Deleted console.log
5. Deleted "undefined"

File: tests/Global_settings/constant.ts [hunk 6]
```diff
@@ -0,0 +1 @@
+export let postAndGetPet = 'https://petstore.swagger.io/v2/pet/';
```
Comment (siarheima-papaya) [inline tests/Global_settings/constant.ts:1]: Rename to 'baseUrl'

File: tests/Api_tests/get.spec.ts [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+   });
+
+
+test('Get', async () => {
+    const response = await axios.get(`${postAndGetPet}${petId}`);
+    expect(response.status).toBe(200);
+});
+
+
+test.afterAll(async () => {
```
Comment (siarheima-papaya) [inline tests/Api_tests/get.spec.ts:45]: Move afterAll() after beforeAll

File: tests/Api_tests/put.spec.ts [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+    {
+      "id": 0,
+      "name": "string"
+    }
+  ],
+  "status": "available"
+ };
+
+
+test('Post', async () => {
```
Comment (siarheima-papaya) [inline tests/Api_tests/put.spec.ts:35]: beforeAll

File: tests/Api_tests/put.spec.ts [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+        ],
+        "tags": [
+          {
+            "id": 0,
+            "name": "string"
+          }
+        ],
+        "status": "available"}
+    });
+    expect(response.status).toBe(200);
```
Comment (siarheima-papaya) [inline tests/Api_tests/put.spec.ts:63]: expect(res.data.category.name).toEqual(updatedFirstName)
expect(res.data.name).toEqual(updatedSecondName)

File: tests/Api_tests/get.spec.ts [hunk 10]
```diff
@@ -41,7 +41,6 @@ test.afterAll(async () => {
 
 test('Get', async () => {
     const response = await axios.get(`${baseUrl}${petId}`);
-    expect(response.data.id).toEqual(randomId);
     expect(response.data.category.name).toEqual(randomFirstName);
     expect(response.data.name).toEqual(randomSecondName);
```
Comment (siarheima-papaya) [inline tests/Api_tests/get.spec.ts:45]: Please, try to compare objects using `expect.objectContaining`. As I remember GET request return the same object as body for POST
```
{
  "id": 0,
  "category": {
    "id": 0,
    "name": "string"
  },
  "name": "doggie",
  "photoUrls": [
    "string"
  ],
  "tags": [
    {
      "id": 0,
      "name": "string"
    }
  ],
  "status": "available"
}
```

File: playwright.config.ts [hunk 11]
```diff
@@ -10,7 +10,7 @@ import { defineConfig, devices } from '@playwright/test';
  * See https://playwright.dev/docs/test-configuration.
  */
 export default defineConfig({
-  testDir: './automation_cource',
+  testDir: './tests/Api_tests',
```
[the remark this answers is not in the document]
Comment (siarheima-papaya) [reply playwright.config.ts:13]: I see 2 directory: Api_tests and api_tests. 
Let's simplify renaming: create new directory `api_tests_pet_strore` -> move tests in it -> remove Api_tests and api_tests

File: tests/Api_tests/post.spec.ts [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+  await axios.delete(`${baseUrl}${petId}`);
+});
+
+test('Post', async () => {
+    const response = await axios.post(baseUrl, postData);
+    expect(response.status).toBe(200);
+    expect(response.data.id).toEqual(randomIdNumber);
+    expect(response.data.category.name).toEqual(randomFirstName);
+    expect(response.data.name).toEqual(randomSecondName);
+    petId = response.data.id;
```
Comment (siarheima-papaya) [inline tests/Api_tests/post.spec.ts:43]: better to move this line directly after POST request
```
const response = await axios.post(baseUrl, postData);
petId = response.data.id;
...
```

File: tests/Api_tests/put.spec.ts [hunk 13]
```diff
@@ -0,0 +1,77 @@
...
+
+test.afterAll(async () => {
+    await axios.delete(`${baseUrl}${petId}`);
+});
+
+
+test('PutRequest', async () => {
+    const response = await axios.put(baseUrl, 
+    { 
+      data: {
```
Comment (siarheima-papaya) [inline tests/Api_tests/put.spec.ts:49]: remove key 'data'. You should pass in just object, without `data:`
And then your test should works fine and response will contain names. So, it means that you can remove `test ('Get',....`

Screenshot 2023-10-03 at 16 13 16

**Comment by Igor-Arena** [comment]
1. Deleted folder Api_tests
2. Added folder api_tests_pet_store
3. Renamed folder Global_settings to global_settings
4. Deleted "data" keyword from the body in the "Put" test file
5. Deleted "Get" test from the "Put" test file
6. Moved "Petid" variable in the top of the test's body

File: tests/api_tests_pet_store/get.spec.ts [hunk 14]
```diff
@@ -0,0 +1,49 @@
...
+   });
+
+test.afterAll(async () => {
+    await axios.delete(`${baseUrl}${petId}`);
+  });
+
+test('Get', async () => {
+    const response = await axios.get(`${baseUrl}${petId}`);
+    expect(response.data.category.name).toEqual(randomFirstName);
+    expect(response.data.name).toEqual(randomSecondName);
```
Comment (siarheima-papaya) [inline tests/api_tests_pet_store/get.spec.ts:45]: You can replace 44 and 45 lines using only one `expect.objectContaining()`. 
Please take a look at example https://playwright.dev/docs/api/class-genericassertions#generic-assertions-object-containing

File: tests/api_tests_pet_store/post.spec.ts [hunk 15]
```diff
@@ -0,0 +1,46 @@
...
+  await axios.delete(`${baseUrl}${petId}`);
+});
+
+test('Post', async () => {
+    const response = await axios.post(baseUrl, postData);
+    petId = response.data.id;
+    expect(response.status).toBe(200);
+    expect(response.data.id).toEqual(randomIdNumber);
+    expect(response.data.category.name).toEqual(randomFirstName);
+    expect(response.data.name).toEqual(randomSecondName);
```
Comment (siarheima-papaya) [inline tests/api_tests_pet_store/post.spec.ts:43]: The same. You can replace by only one expect.objectContaining().

**Comment by Igor-Arena** [comment]
1. Added expect.objectContaining
2. Added toMatchObject

File: tests/api_tests_pet_store/delete.spec.ts [hunk 16]
```diff
@@ -0,0 +1,42 @@
...
+
+
+let randomFirstName = faker.person.firstName();
+let randomSecondName = faker.person.lastName();
+let randomId = faker.number.int(10);
+let petId: number;
+
+
+let postData = {
+  "id": randomId,
```
Comment (siarheima-papaya) [inline tests/api_tests_pet_store/delete.spec.ts:14]: I just let you know, that it's possible to use key without quotes. 
E.q.
```
{
  id: randomId,
.....
 };
```

File: tests/api_tests_pet_store/put.spec.ts [hunk 17]
```diff
@@ -0,0 +1,76 @@
...
+    }
+  ],
+  "status": "available"
+ };
+
+
+test.beforeAll(async () => {
+    const response = await axios.post(baseUrl, postData);
+    petId = response.data.id;
+    expect(response.status).toBe(200);
```
Comment (siarheima-papaya) [inline tests/api_tests_pet_store/put.spec.ts:38]: Please, remove assertion from beforeAll()

**Comment by Igor-Arena** [comment]
Removed assertion from beforeAll in "Put" test

**Review by siarheima-papaya — APPROVED** [review]

**Pull request merged**